.github/workflows/tests.yml

```diff
@@ -0,0 +1,74 @@
+name: "tests"
+
+on: ["pull_request", "push"]
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: "7.1"
+          - php-version: "7.2"
+          - php-version: "7.3"
+          - php-version: "7.4"
+          - php-version: "8.0"
+          - php-version: "8.1"
+          - php-version: "8.1"
+            prefer: lowest
+
+    steps:
+      - name: Start MySQL
+        env:
+          MYSQL_PWD: root
+        run: |
+          sudo systemctl start mysql.service
+          mysql -uroot -e "create database odm";
+
+      - name: Start PostgreSQL
+        run: |
+          sudo systemctl start postgresql.service
+          pg_isready
+          sudo -u postgres psql -c "create user odm WITH PASSWORD 'odm';"
+          sudo -u postgres psql -c "create database odm OWNER odm;"
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: mbstring, xml, ctype, iconv, intl, pdo_mysql, pdo_pgsql
+
+      - name: Get composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ matrix.prefer }}-
+          restore-keys: ${{ runner.os }}-composer-${{ matrix.prefer }}-
```

Comment:
this restore key is wrong (or the cache key), as the composer.json hash is not at the end

Reply:
Fixed in https://github.com/dunglas/doctrine-json-odm/pull/111.
The problem comes from a snipped I copied from the README of setup-php: https://github.com/shivammathur/setup-php#cache-composer-dependencies

Reply:
their example is OK. their restore key is a prefix of their key. But you added `-${{ matrix.prefer }}` at the end instead of before the hash